packages/marshal/marshal.js

```diff
@@ -296,7 +296,31 @@ function makeReviverIbidTable(cyclePolicy) {
   });
 }
 
-export function makeMarshal(serializeSlot, unserializeSlot) {
+const identityFn = x => x;
+
+export function makeMarshal(
+  convertValToSlot = identityFn,
+  convertSlotToVal = identityFn,
+) {
+  function serializeSlot(val, slots, slotMap) {
+    let slotIndex;
+    if (slotMap.has(val)) {
+      slotIndex = slotMap.get(val);
+    } else {
+      slotIndex = slots.length;
```

Comment:
My taste/style would be to move this `slotsIndex=` line down to just before the `slots.push` line, to make it clear that managing this list is entirely independent of calling `convertValToSlot`. (Also, if for some strange reason `convertValToSlot` *did* recurse and provoke more calls to `serializeSlot`, I think this could produce a TOCTTOU bug).